cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -263,7 +264,8 @@ impl<State> CoreEnvironment<State> {
 
         let service_config_path = maybe_make_service_config_file(flox, &lockfile)?;
 
-        let outputs = BuildEnvNix::new("/tmp/flox-temp-roots/").build(
+        let tempdir = TempDir::new().map_err(CoreEnvironmentError::CreateTempdir)?;
```

Comment:
blocking suggestion: hold onto the tempdir until we finish building the environment or persist the tempdir. The whole point of creating this directory is to keep gcroots around until we finish the build and prevent gc via the environment's gcroot

Reply:
The tempir lives as long as `BuildEnvNix` does, so it lives until after the build is done, or am I misunderstanding your comment?

Reply:
Ah sorry thought this was further down, nvm